### PR TITLE
Code review best practices+ bad practices and styling

### PR DESCRIPTION
**Pull request by rmacken98**
Added Code Review best practices
Added quick links + references
Added Images to best practices
Fixed spelling mistake in code review bad practices.

### CODE REVIEW

**Review by Lordjiggyx — CHANGES_REQUESTED** [review]
Code review Best Practices: second image not rendering
Bad Parctices : code review quick link not working

**Comment by NicBoi** [comment]
In CRBP.md:
- I suggest renaming this file to a similar style to all other file names
- There are some spelling errors that need to be fixed
- Images should be moved from the Bad_Practices folder into this topic's folder and referenced, also update alternative text
- Update quick links. Home page should point to README.md file

CodeReviewBestPractices.md:
- There are some spelling errors that need to be fixed
- image002.png would be better to rename to a more appropriate name

Overall in quick links "./" (different from "../") should be removed as it has no effect.

Also the code review topic seems to contain too much information (a bit more than 2000 words), it should be between 500 to 1000 words

**Comment by NicBoi** [comment]
Too many features are being dealt with in this branch.
For now I think the PR title should be changed to something more appropriate, like "Code review best + bad practices and styling"

**Comment by rmacken98** [comment]
> Too many features are being dealt with in this branch.
> For now I think the PR title should be changed to something more appropriate, like "Code review best + bad practices and styling"

I'll remove some of the features from this branch such as the references and quick links and address them in a separate PR.
I will also address the revising the length of this topic in a separate PR. If that is suitable?

**Comment by NicBoi** [comment]
> > Too many features are being dealt with in this branch.
> > For now I think the PR title should be changed to something more appropriate, like "Code review best + bad practices and styling"
> 
> I'll remove some of the features from this branch such as the references and quick links and address them in a separate PR.
> I will also address the revising the length of this topic in a separate PR. If that is suitable?

For now you can keep everything here, as there might be too much work to be done in separating everything. But just to be aware in the future @rmacken98

**Comment by Lordjiggyx** [comment]
Looks good for merge

File: CodeReviews/CRBP.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Bad Practices - Code Reviwes
+# Bad Practices - Code Reviews
 
 * **Review Code Not People**  
 When reviewing others code it is best to remember that it is not a competition to see who can code better. Software developers are often proud of thier solutions and take criticism personally on occasion. Reviews help us see how others thought process occurs and feedback gievn should be constructive uplifing and insightful to help others program more efficiently. Be careful to always provide feedback that does not question the person but questions how and why they created a solution the way they did.
```
Comment (NicBoi) [inline CodeReviews/CRBP.md]:
```suggestion
When reviewing others code it is best to remember that it is not a competition to see who can code better. Software developers are often proud of their solutions and take criticism personally on occasion. Reviews help us see how others thought process occurs and feedback given should be constructive uplifting and insightful to help others program more efficiently. Be careful to always provide feedback that does not question the person but questions how and why they created a solution the way they did.
```
Minor spelling errors

File: CodeReviews/CRBP.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ The team should decide on how they want to style their code, your teams code sho
 
 
 * **Meaningful comments**  
-If a person makes an error while programming it is almost certain that this error has been repeated across in other parts. When reviewing others code do not point out every occurence of the error instead leave a detailed not explaining the overall issue. It is essential to reply to all comments in order to ensure that the review process is actually being completed.
+If a person makes an error while programming it is almost certain that this error has been repeated across in other parts. When reviewing others code do not point out every occurrence of the error instead leave a detailed not explaining the overall issue. It is essential to reply to all comments in order to ensure that the review process is actually being completed.
 
 <p align="center">
 <img src="https://miro.medium.com/max/1400/1*UbeT74oIf3JlX1YRu90EKA.jpeg" alt="T-Shirt Size Image" width="400" center>
```
Comment (NicBoi) [inline CodeReviews/CRBP.md:18]: Would be better to reference the images from within the github repo, as done for other topics. Also to update the alternative text

File: CodeReviews/CRBP.md [hunk 3]
```diff
@@ -30,6 +30,6 @@ Feedback is useless if it is not taken to heart. When the review rounds and disc
 [How to sabotage a code review](https://techbeacon.com/app-dev-testing/how-sabotage-code-review)
 
 ## Quick Links
-[Code Review](../../CodeReviews/CodeReviews.md)
+[Code Review](CodeReviews.md)
 
-[Home Page](../../ProjectPlan.md)
\ No newline at end of file
+[Home Page](../../ProjectPlan.md)
```
Comment (NicBoi) [inline CodeReviews/CRBP.md]:
```suggestion
[Home Page](../README.md)
```
Home page should point to the README file

File: CodeReviews/CodeReviewBestPractices.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
+# Best Practices 
+
+ <p align="center">
+<img src="Images/cleancodereview.png" alt="Defects Density vs Lines of code per hour" width="400">
+</p>
+
+### Communicate clear goals and expectations of reviewers  
+ To ensure that reviews are consistent Reviewers should be given a checklist of what to look for during a review. The checklist should contain the following critrea of what they should be looking for as reviewers :
```
Comment (NicBoi) [inline CodeReviews/CodeReviewBestPractices.md]:
```suggestion
 To ensure that reviews are consistent Reviewers should be given a checklist of what to look for during a review. The checklist should contain the following criteria of what they should be looking for as reviewers :
```
Minor spelling error

File: CodeReviews/CodeReviewBestPractices.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+ #### Design 
+* When looking at the design of the code the reviewer should be asking themselves these questions :
+
+ * Does the change belong in the code base or in a library.
+ * How well does it intergrate with the rest of the system.
```
Comment (NicBoi) [inline CodeReviews/CodeReviewBestPractices.md]:
```suggestion
 * How well does it integrate with the rest of the system.
```
Minor spelling error

File: CodeReviews/CodeReviewBestPractices.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+ * Does the code carry out it's intended function?
+ * Is it not only good for the users but also other developers who will have to work with this area of the code base in the future?
+ * What are the edge cases?, Think like a user and make sure there are no bugs or concurrency issues.
+
+ #### Structure
+* Are there any structural errors in the code such as dead code, algorithm bugs etc.? (this task can usually be automated)
+
+ #### Style 
+* Is the code consistent with the style guide? 
+ This is important for maintaining consistency and helps with readibility across the project.
```
Comment (NicBoi) [inline CodeReviews/CodeReviewBestPractices.md]:
```suggestion
 This is important for maintaining consistency and helps with readability across the project.
```
Minor spelling error

File: CodeReviews/CodeReviewBestPractices.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+* Do unit or end-to-end tests need to be updated/created?
+
+
+### Speed of a code review and how long should be spent on a given review
+ 
+ #### 60-90 minutes max per review
+ A code reviewer should take the time to carry out a slow and thorough review. However this should take no more than 60-90 minutes according to study's carried out by IBM and others it has been shown that after 60 minutes reviewers tend to get tired and stop finding additional defects/issues with the code.
+
+ <p align="center">
+<img src="Images/image002.gif" alt="Defects Density vs Lines of code per hour" width="400">
```
Comment (NicBoi) [inline CodeReviews/CodeReviewBestPractices.md:44]: Would be nice to rename image to a more appropriate name

File: CodeReviews/CodeReviewBestPractices.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+ If you are not convinced by their argument for not accepting your review ask them for further explanation on why they believe they're correct. If you are still not convinced by their argument believing your suggestion and the additional work involved is worth the benefit it will provide to the project continue to push for the developer to accept your review and make the changes.
+
+### Include Everyone in the review process
+Including everyone in the review process no matter what their seniority as a developer is helps promote collaboration and relationships between the team members enhancing the team's cohesion. Also as mentioned before in code review benefits knowing that every other team member will be looking at your code will encourage you to make sure code is of a higher standard when it's being put up for a review.
+
+
+ ## Quick Links
+  * [Home Page](../README.md)
+  * [Code Review Home Page](./CodeReviews.md)
+  * [Bad Practices](./CRBP.md)
```
Comment (NicBoi) [inline CodeReviews/CodeReviewBestPractices.md]:
```suggestion
  * [Code Review Home Page](CodeReviews.md)
  * [Bad Practices](CRBP.md)
```
Unnecessary "./", it has no effect

File: CodeReviews/CodeReviews.md [hunk 9]
```diff
@@ -16,16 +16,29 @@ A code review inolves checking for things like :
...
 #### Code reviews enable time off
  * Since knowledge is shared across the project amongst each team member through code reviews. Each member of the team will have have an understanding of what their peers are working on meaning there isn't a reliance on a single team member for a piece of code and that any other team member should be able to take over their work. Allowing for team members to take time off when needed.
+ 
 
 
- 
+## Quick Links
+  * [Home Page](../README.md)
+  * [Best Practices](./CodeReviewBestPractices.md)
+  * [Bad Practices](./CRBP.md)
```
Comment (NicBoi) [inline CodeReviews/CodeReviews.md]:
```suggestion
  * [Best Practices](CodeReviewBestPractices.md)
  * [Bad Practices](CRBP.md)
```
Unnecessary "./", as it has no effect

**Review by NicBoi — CHANGES_REQUESTED** [review]

**Review by seanymcelroy — APPROVED** [review]
After following changes requested, this is good to merge!

File: CodeReviews/CodeReviewBestPractices.md (hunk 7 above)
Comment (rmacken98) [reply CodeReviews/CodeReviewBestPractices.md:44]: Will be renamed in pr revising the length of the topic.

**Pull request merged**